src/python/pants/backend/codegen/tasks/simple_codegen_task.py

```diff
@@ -78,6 +103,31 @@ def codegen_targets(self):
     """
     return self.context.targets(self.is_gentarget)
 
+  @classmethod
+  def _codegen_strategy_for_name(cls, name):
+    return {
+      'global': SimpleCodegenTask.GlobalCodegenStrategy,
+      'isolated': SimpleCodegenTask.IsolatedCodegenStrategy,
+    }[name]
+
+  @property
+  def codegen_strategy(self):
+    """Returns the codegen strategy object used by this codegen.
+
+    This is controlled first by the forced_codegen_strategy property, then by user-specified
+    options if the former returns None.
+
+    If you just want the name ('global' or 'isolated') of the strategy, use codegen_strategy.name.
+
+    :return SimpleCodegenTask.CodegenStrategy: the codegen strategy object.
+    """
+    strategy = self.forced_codegen_strategy
+    if strategy is None:
+      strategy = self.get_options().strategy
+    if self._codegen_strategy is None or self._codegen_strategy.name != strategy:
```

Comment:
It was mostly that I wanted to make sure I was testing the logic that decided which code-gen strategy to use.